src/poetry/core/poetry.py

```diff
@@ -48,3 +55,35 @@ def local_config(self) -> dict[str, Any]:
 
     def get_project_config(self, config: str, default: Any = None) -> Any:
         return self._local_config.get("config", {}).get(config, default)
+
+    @property
+    def build_system_dependencies(self) -> list[Dependency]:
+        if self._build_system_dependencies is None:
+            build_system = self.pyproject.build_system
+            self._build_system_dependencies = []
+
+            for requirement in build_system.requires:
+                dependency = None
+                try:
+                    dependency = Dependency.create_from_pep_508(requirement)
+                except ValueError:
+                    # PEP 517 requires can be path if not PEP 508
+                    path = Path(requirement)
+
+                    with contextlib.suppress(OSError):
+                        # suppress OSError for compatibility with Python < 3.8
+                        # https://docs.python.org/3/library/pathlib.html#methods
```

Comment:
There's no need to take care for Python <3.8. Minimum required Python version is already. 3.9.

Reply:
True. I just ported it over from the old PR. Will tidy this up today.